include/net/nrf_cloud_log.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_LOG_H_
+#define NRF_CLOUD_LOG_H_
+
+/** @file nrf_cloud_log.h
+ * @brief Module to provide nRF Cloud logging support to nRF9160 SiP.
+ */
+
+#include <zephyr/logging/log.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct nrf_cloud_rest_context;
+
+/**
+ * @defgroup nrf_cloud_log nRF Cloud Log
+ * @{
+ */
+
+/**
+ * @brief Initialize the nRF Cloud logging subsystem. Call after
+ * the date_time library is ready.
+ */
+void nrf_cloud_log_init(void);
+
+/**
+ * @brief Set REST context for logging subsystem.
+ *
+ * Tell REST-based logger the REST context and device_id for later
+ * use when outputting a log. This is used by the nRF Cloud logs backend
+ * when CONFIG_NRF_CLOUD_LOG_BACKEND=y.
+ *
+ * @param[in] ctx REST context to use.
+ * @param[in] dev_id Device ID to send message on behalf of.
+ */
+void nrf_cloud_log_rest_context_set(struct nrf_cloud_rest_context *ctx, const char *dev_id);
+
+/**
+ * @brief Set the logging level.
+ *
+ * Set the log level, but do not change whether the nRF Cloud logging backend
+ * is enabled.
+ *
+ * Log messages with a log level larger than this will not be sent.
+ * For example, if log_level is LOG_LEVEL_WRN (2), then
+ * LOG_LEVEL_WRN (2) and LOG_LEVEL_ERR (1) messages will be sent, and
+ * LOG_LEVEL_INF (3) and LOG_LEVEL_DBG (4) will not.
+ * Set log_level to LOG_LEVEL_NONE (0) to disable logging.
+ * See zephyr/include/zephyr/logging/log_core.h.
+ *
+ * @param level The log level.
+ */
+void nrf_cloud_log_level_set(int level);
+
+/**
+ * @brief Enable or disable logging to the cloud.
+ *
+ * When CONFIG_NRF_CLOUD_LOG_BACKEND=y, will enable or disable the log backend,
+ * without changing the log level. Disable the log backend to conserve power by
+ * eliminating unnecessary processing of log messages.
+ *
+ * When CONFIG_NRF_CLOUD_LOG_DIRECT=y, the log enable state determines whether
+ * calls to nrf_cloud_log_send() or nrf_cloud_rest_log_send() will succeed.
+ * This will conserve power by not transmitting log messages when not needed.
```

Comment:
The Zephyr log system has its own buffer, so that will continue to do so, but once a message gets passed to `logger_process` in `nrf_cloud_log_backend.c`, it is ignored if `enable = false`, so will not have any opportunity to be placed in the ring buffer.